service/controller/v23/adapter/guest_auto_scaling_group.go

```diff
@@ -49,8 +49,18 @@ func (a *GuestAutoScalingGroupAdapter) Adapt(cfg Config) error {
 	a.ASGMinSize = minWorkers
 	a.ASGType = key.KindWorker
 	a.ClusterID = key.ClusterID(cfg.CustomObject)
-	a.MaxBatchSize = workerCountRatio(currentDesiredMinWorkers, asgMaxBatchSizeRatio)
-	a.MinInstancesInService = workerCountRatio(currentDesiredMinWorkers, asgMinInstancesRatio)
+	a.MaxBatchSize = strconv.Itoa(workerCountRatio(currentDesiredMinWorkers, asgMaxBatchSizeRatio))
+
+	minInstancesInService := workerCountRatio(currentDesiredMinWorkers, asgMinInstancesRatio)
+	if minWorkers == 1 && maxWorkers == 1 && minInstancesInService == maxWorkers {
```

Comment:
```suggestion
	if minWorkers == 1 && maxWorkers == 1 && minInstancesInService == 1 {
```

Reply:
I also think we can remove the third part of the condition all together. Is it possible that minInstancesInService  isn't 0 or 1 in that case?

Reply:
It currently is always 1 when this is the case, but yeah the condition is not really needed. I removed the third part and will test again to make sure that everything works correctly.